src/components/Map/EditableMapV2.tsx

```diff
@@ -41,24 +43,23 @@ export default function EditableMap({
   editableBoundary,
   onEditableBoundaryChanged,
   onUndoRedoReadOnlyBoundary,
+  overridePopupInfo,
+  popupRenderer,
   readOnlyBoundary,
   setMode,
   style,
 }: EditableMapProps): JSX.Element {
   const { mapId, refreshToken, token } = useMapboxToken();
+  const [editMode, setEditMode] = useState<MapEditorMode>();
   const [firstVisible, setFirstVisible] = useState<boolean>(false);
+  const [interactiveLayerIds, setInteractiveLayerIds] = useState<string[]>([]);
+  const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
   const containerRef = useRef(null);
   const mapRef = useRef<MapRef | null>(null);
   const visible = useIsVisible(containerRef);
   const theme = useTheme();
   const [mapViewStyle, onChangeMapViewStyle] = useMapViewStyle();
-
-  useEffect(() => {
-    // `firstVisible` detects when the box containing the map is first visible in the viewport. The map should only be
-    // rendered if `firstVisible` is true. This accounts for cases in which the map is initially rendered hidden, and
-    // is improperly resized when it first becomes visible.
-    setFirstVisible((fv) => fv || visible);
-  }, [visible]);
+  let destroying = false;
```

Comment:
Should this be hooked up to state?

Reply:
I lifted this from [here](https://github.com/terraware/terraware-web/blob/main/src/components/Map/Map.tsx#L439), upon debug/experimentation the map destroy is called after the component is unmounted (the state is no longer accessible). This is hacky, I can remove the whole destroying logic and see if anything barfs when I clean things up?

Reply:
Removed this destroy logic for now, not sure it's needed.